aiozk/connection.py

```diff
@@ -168,27 +172,39 @@ def handle_write(write_future):
             else:
                 f.set_result((zxid, response))
 
+    async def _read(self, size=-1):
+        remaining_size = size
+        payload = b''
+        end_time = time() + self.read_timeout
+        while remaining_size and (time() < end_time):
+            chunk = await self.reader.read(remaining_size)
+            payload += chunk
```

Comment:
This kind of antipattern, so it would be better to have it written like this:
```python
payload = []
payload.append(chunk)
return b''.join(payload)
```